frontend/src/components/PromotionList.tsx

```diff
@@ -29,8 +29,8 @@ export default function PromotionList({
   };
 
   useEffect(() => {
-    const { filter, sort, promotions } = state;
-    const filteredPromotions = filterPromotions(promotions, filter);
+    const { filter, sort, data } = state;
+    const filteredPromotions = filterPromotions(data, filter);
```

Comment:
To double-check, `filter, sort, data` comes from the reducer?

Reply:
They come from the state! The reducer just makes any updates to the state whenever `dispatch` is called

Reply:
Ohhh I see okay that makes sense thanks!